src/storage/src/compaction/writer.rs

```diff
@@ -0,0 +1,467 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_query::logical_plan::{DfExpr, Expr};
+use datafusion_common::ScalarValue;
+use datafusion_expr::{BinaryExpr, Operator};
+
+use crate::chunk::{ChunkReaderBuilder, ChunkReaderImpl};
+use crate::error;
+use crate::schema::RegionSchemaRef;
+use crate::sst::{AccessLayerRef, FileHandle};
+
+/// Builds an SST reader that only reads rows within given time range.
+pub(crate) async fn build_sst_reader(
+    schema: RegionSchemaRef,
+    sst_layer: AccessLayerRef,
+    files: &[FileHandle],
+    lower_sec_inclusive: i64,
+    upper_sec_exclusive: i64,
+) -> error::Result<ChunkReaderImpl> {
+    let ts_col_name = schema
```

Comment:
We assume that the time index column is not changed here, but it may happen in future. So we have to process this situation. Can we add a `TODO` here and create an issue to track it?